lib/streamlit/web/server/media_file_handler.py

```diff
@@ -0,0 +1,130 @@
+# Copyright 2018-2022 Streamlit Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from typing import Optional
+from urllib.parse import quote, unquote_plus
+
+import tornado.web
+
+from streamlit.logger import get_logger
+from streamlit.runtime.media_file_manager import (
+    _get_extension_for_mimetype,
+    media_file_manager,
+    MediaFileType,
+)
+from streamlit.string_util import generate_download_filename_from_title
+from streamlit.web.server import allow_cross_origin_requests
+
+LOGGER = get_logger(__name__)
+
+
+class MediaFileHandler(tornado.web.StaticFileHandler):
+    def set_default_headers(self) -> None:
+        if allow_cross_origin_requests():
+            self.set_header("Access-Control-Allow-Origin", "*")
+
+    def set_extra_headers(self, path: str) -> None:
+        """Add Content-Disposition header for downloadable files.
+
+        Set header value to "attachment" indicating that file should be saved
+        locally instead of displaying inline in browser.
+
+        We also set filename to specify the filename for downloaded files.
+        Used for serving downloadable files, like files stored via the
+        `st.download_button` widget.
+        """
+        media_file = media_file_manager.get(path)
+
+        if media_file and media_file.file_type == MediaFileType.DOWNLOADABLE:
+            file_name = media_file.file_name
+
+            if not file_name:
+                title = self.get_argument("title", "", True)
+                title = unquote_plus(title)
+                filename = generate_download_filename_from_title(title)
+                file_name = (
+                    f"{filename}{_get_extension_for_mimetype(media_file.mimetype)}"
```

Comment:
since new class property `extension` added `f"{filename}{media_file.extension}" could be used here

Reply:
Yes, good call!